src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Implementation/Resource/FindAzureResourceCmdlet.cs

```diff
@@ -110,19 +125,20 @@ public sealed class FindAzureResourceCmdlet : ResourceManagerCmdletBase
         [ValidateNotNullOrEmpty]
         public string ODataQuery { get; set; }
 
+        [Parameter(ParameterSetName = FindAzureResourceCmdlet.ListResourcesByTagObjectParameterSet, Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = "The tag filter for the OData query. The expected format is @{tagName=$null} or @{tagName = 'tagValue'}.")]
+        public Hashtable Tag { get; set; }
+
         /// <summary>
         /// Gets or sets the tag name.
         /// </summary>
-        [Parameter(ParameterSetName = FindAzureResourceCmdlet.ListResourcesParameterSet, Mandatory = false, HelpMessage = "The name of the tag to query by.")]
-        [Parameter(ParameterSetName = FindAzureResourceCmdlet.MultiSubscriptionListResourcesParameterSet, Mandatory = false, HelpMessage = "The name of the tag to query by.")]
```

Comment:
Hi Cormac,

  This change is done intentionally to fix a bug that allowed illegal combination of parameters.
So only TagValue and Top should be allowed with TagName.
Only Top should be allowed with the Tag parameter.

Our service does not support other combinations. Today the user is allowed to specify other combinations but it results in errors like below:
1.	Find-AzureRmResource -TagName "TestTag" -TagValue "testVal1"  -ResourceNameContains "abc"
Gives the following error:
Find-AzureRmResource : InvalidFilterInQueryString : Invalid $filter 'tagName EQ 'TestTag' AND tagValue EQ 'testVal1' AND substringof('abc', name)' specified in the query string.

2.	Find-AzureRmResource -TagName "TestTag" -TagValue "testVal1"  -TenantLevel
Gives the following error:
Find-AzureRmResource : Parameter set cannot be resolved using the specified named parameters.

3.	Find-AzureRmResource -Tag @{TestTag="testVal1"} -ResourceType "Microsoft.Web/sites"
Gives the following error:
Find-AzureRmResource : InvalidFilterInQueryString : Invalid $filter 'resourceType EQ 'Microsoft.Web/sites' AND tagName EQ 'TestTag' AND tagValue EQ 'testVal1'' specified in the query string.

With this parameter set change I am attempting to fix this bug.